compiler/crates/relay-compiler/src/artifact_content/content.rs

```diff
@@ -178,11 +180,24 @@ pub fn generate_operation(
     fragment_locations: &FragmentLocations,
 ) -> Result<Vec<u8>, FmtError> {
     let mut request_parameters = build_request_params(normalization_operation);
-    if id_and_text_hash.is_some() {
-        request_parameters.id = id_and_text_hash;
-    } else {
-        request_parameters.text = text.clone();
-    };
+    match (id_and_text_hash, &project_config.persist) {
+        (
+            Some(_),
+            Some(PersistConfig::Remote(RemotePersistConfig {
+                include_query_text, ..
+            }))
+        ) if *include_query_text => {
+            request_parameters.id = id_and_text_hash;
+            request_parameters.text = text.clone();
+        }
+        (Some(_), _) => {
+            request_parameters.id = id_and_text_hash;
+        }
+        (None, _) => {
+            request_parameters.text = text.clone();
+        }
+    }
```

Comment:
I find these tuple matches really hard to follow. I _think_ this is the same logic? (See my comment elsewhere for the `include_query_text` method.

```suggestion
    if id_and_text_hash.is_some() {
        request_parameters.id = id_and_text_hash;
        if project_config
            .persist
            .as_ref()
            .map_or(false, |config| config.include_query_text())
        {
            request_parameters.text = text.clone();
        }
    } else {
        request_parameters.text = text.clone();
    }
```